cmd/hypersdk-cli/key_set.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package main
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"github.com/ava-labs/hypersdk/auth"
+	"github.com/ava-labs/hypersdk/crypto/ed25519"
+)
+
+var keyGenerateCmd = &cobra.Command{
+	Use:   "generate",
+	Short: "Generate a new key",
+	RunE: func(cmd *cobra.Command, _ []string) error {
+		newKey, err := ed25519.GeneratePrivateKey()
+		if err != nil {
+			return fmt.Errorf("failed to generate key: %w", err)
+		}
+
+		return checkAndSavePrivateKey(hex.EncodeToString(newKey[:]), cmd)
+	},
+}
+
+var keySetCmd = &cobra.Command{
+	Use:   "set",
+	Short: "Set the private ED25519 key",
+	RunE: func(cmd *cobra.Command, _ []string) error {
+		// read directly from the flag instead of calling getConfigValue
+		keyString, err := cmd.Flags().GetString("key")
+		if err != nil {
+			return fmt.Errorf("failed to get key: %w", err)
+		}
+		if keyString == "" {
+			return errors.New("--key is required")
+		}
+
+		return checkAndSavePrivateKey(keyString, cmd)
+	},
+}
+
+func checkAndSavePrivateKey(keyString string, cmd *cobra.Command) error {
```

Comment:
could we reverse the order of arguments here? We seem to use `cmd` as the first argument everywhere else.